### PR TITLE
[CELEBORN-1766] Add detail metrics about fetch chunk

### PR DESCRIPTION
**Pull request by FMX**
### What changes were proposed in this pull request?
1. Add histogram
2. Collect critical metrics about fetch chunk

### Why are the changes needed?
1. To find out IO pattern of fetch chunk
2. To have detail metrics about fetch chunk time 

### Does this PR introduce _any_ user-facing change?
NO.

### How was this patch tested?
GA and cluster.

截屏2024-12-09 15 42 50
截屏2024-12-09 15 42 56

### CODE REVIEW

File: common/src/main/java/org/apache/celeborn/common/network/TransportContext.java [hunk 1]
```diff
@@ -111,6 +111,17 @@ public TransportContext(
     this(conf, msgHandler, closeIdleConnections, null, enableHeartbeat, source);
   }
 
+  public TransportContext(
+      TransportConf conf,
+      BaseMessageHandler msgHandler,
+      boolean closeIdleConnections,
+      boolean enableHeartbeat,
+      AbstractSource source,
+      boolean collectFetchChunkDetailMetrics) {
```
Comment (turboFei) [inline common/src/main/java/org/apache/celeborn/common/network/TransportContext.java:120]: seems this variable is not used here

File: assets/grafana/celeborn-dashboard.json [hunk 2]
```diff
@@ -4386,25 +4954,50 @@
...
-            "overrides": []
+            "overrides": [
+              {
+                "__systemRef": "hideSeriesFrom",
+                "matcher": {
+                  "id": "byNames",
+                  "options": {
+                    "mode": "exclude",
+                    "names": [
+                      "metrics_ActiveChunkStreamCount_Value {__name__=\"metrics_ActiveChunkStreamCount_Value\", exported_instance=\"10.230.216.22:9096\", instance=\"core-1-3:9096\", job=\"RSS\", role=\"worker\"}"
```
Comment (turboFei) [inline assets/grafana/celeborn-dashboard.json:4975]: could you revert the unnecessary changes?

File: common/src/main/java/org/apache/celeborn/common/network/TransportContext.java (hunk 1 above)
Comment (FMX) [reply common/src/main/java/org/apache/celeborn/common/network/TransportContext.java:120]: updated

File: assets/grafana/celeborn-dashboard.json (hunk 2 above)
Comment (turboFei) [reply assets/grafana/celeborn-dashboard.json]: could you take a look at https://github.com/FMX/celeborn/pull/8?

Here is the commit that only add new metrics, https://github.com/FMX/celeborn/pull/8/commits/f666e59f35a4f76699d6c72ccdcb408e5364a642

**Review by RexXiong — APPROVED** [review]

**Comment by RexXiong** [comment]
Merge to main(v0.6.0)
